src/App.js

```diff
@@ -1,9 +1,63 @@
-function App () {
-  return (
-    <div>
-      <h1>Hello, world!</h1>
-    </div>
-  );
-};
+import React, { useState } from "react";
+import birdData from "./data/birds";
+import Cart from "./Components/Cart";
+import Checkout from "./Components/Checkout";
+import Display from "./Components/Display";
+import "./App.css"
+
+class App extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      adoptedBirds: [],
+    };
+  }
+
+  adoptHandler = (id, name, amount) => {
+    let copyOfAdoptedBirds = [...this.state.adoptedBirds];
+    copyOfAdoptedBirds.push({ id, name, amount });
+    this.setState({
+      adoptedBirds: copyOfAdoptedBirds,
+    });
+  };
+
+  checkOutHandler = (event) => {
+    event.preventDefault();
+    if (this.state.adoptedBirds.length > 0) {
+      alert("You have adopted birds. Thank you!");
+      this.setState({
+        adoptedBirds: [],
+      });
+    } else {
+      alert("You haven't adopted any birds yet!");
+    }
+  };
+
+  removeItemHandler = (id) => {
+    let copyOfAdoptedBirds = [...this.state.adoptedBirds];
+    for (let i = 0; i < copyOfAdoptedBirds.length; i++) {
+      if (copyOfAdoptedBirds[i].id === id) {
+        copyOfAdoptedBirds.splice(i, 1);
+        break;
+      }
+    }
```

Comment:
In your implementation of this function you won't necessarily delete the burd that was clicked, but the first bird in the array that has that same id. The ids unfortunately are not unique because we can have more than one bird of the same kind and they would have the same id. You can fix this by using the index that you pass in `Cart.js` When you map over the array of birds